pydantic/fields.py

```diff
@@ -554,23 +555,36 @@ def deprecation_message(self) -> str | None:
             return 'deprecated' if self.deprecated else None
         return self.deprecated if isinstance(self.deprecated, str) else self.deprecated.message
 
-    def get_default(self, *, call_default_factory: bool = False) -> Any:
+    @overload
+    def get_default(self, *, call_default_factory: Literal[False]) -> Any: ...
+
+    @overload
+    def get_default(self, *, call_default_factory: Literal[True], validated_data: dict[str, Any]) -> Any: ...
+
+    def get_default(self, *, call_default_factory: bool, validated_data: dict[str, Any] | None = None) -> Any:
         """Get the default value.
 
         We expose an option for whether to call the default_factory (if present), as calling it may
         result in side effects that we want to avoid. However, there are times when it really should
         be called (namely, when instantiating a model via `model_construct`).
 
         Args:
-            call_default_factory: Whether to call the default_factory or not. Defaults to `False`.
+            call_default_factory: Whether to call the default factory or not.
+            validated_data: The already validated data to be passed to the default factory.
 
         Returns:
             The default value, calling the default factory if requested or `None` if not set.
         """
         if self.default_factory is None:
             return _utils.smart_deepcopy(self.default)
         elif call_default_factory:
-            return self.default_factory()
+            if validated_data is None:
+                raise ValueError("'validated_data' must be provided if 'call_default_factory' is True.")
```

Comment:
Is this true? Doesn't it only need to be not `None` when `_fields.takes_validated_data_argument(self.default_factory)` is True?

Reply:
Yes I can probably move the check below

Reply:
Fine by me, feel free to merge then :).